File: app/src/main/java/com/github/batulovandrey/unofficialurbandictionary/adapter/AlphabetAdapter.java
package com.github.batulovandrey.unofficialurbandictionary.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.github.batulovandrey.unofficialurbandictionary.R;

import java.util.List;

/**
 * @author Andrey Batulov on 27/10/2017
 */

public class AlphabetAdapter extends RecyclerView.Adapter<AlphabetViewHolder> {

    private List<String> mAlphabet;
    private AlphabetClickListener mClickListener;

    public AlphabetAdapter(List<String> alphabet, AlphabetClickListener listener) {
        this.mAlphabet = alphabet;
        mClickListener = listener;
    }

    @Override
    public AlphabetViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.alphabet_item, null);
        return new AlphabetViewHolder(view, mClickListener);
    }

    @Override
    public void onBindViewHolder(AlphabetViewHolder holder, int position) {
        String letter = mAlphabet.get(position);
        holder.mAlphabetTextView.setText(letter);
    }

    @Override
    public int getItemCount() {
        return mAlphabet.size();
    }
}

File: app/src/main/java/com/github/batulovandrey/unofficialurbandictionary/adapter/AlphabetAdapter.kt
package com.github.batulovandrey.unofficialurbandictionary.adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.github.batulovandrey.unofficialurbandictionary.R

/**
 * @author Andrey Batulov on 22/12/2017
 */

class AlphabetAdapter(private val mAlphabet: List<String>,
                      private val mClickListener: AlphabetClickListener) : RecyclerView.Adapter<AlphabetViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AlphabetViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.alphabet_item, null)
        return AlphabetViewHolder(view, mClickListener)
    }

    override fun onBindViewHolder(holder: AlphabetViewHolder, position: Int) {
        val letter = mAlphabet[position]
        holder.setAlphabetText(letter)
    }

    override fun getItemCount(): Int {
        return mAlphabet.size
    }
}